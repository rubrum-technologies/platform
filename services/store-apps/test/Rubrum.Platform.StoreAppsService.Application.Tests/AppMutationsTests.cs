using System;
using System.Threading.Tasks;
using CookieCrumble;
using HotChocolate.Types.Relay;
using Microsoft.Extensions.DependencyInjection;
using NSubstitute;
using Rubrum.Platform.StoreAppsService.Apps;
using Shouldly;
using Volo.Abp.Users;
using Xunit;
using static Rubrum.Platform.StoreAppsService.AppTestConstants;

namespace Rubrum.Platform.StoreAppsService;

public class AppMutationsTests : StoreAppsServiceApplicationGraphqlTestBase
{
    private readonly INodeIdSerializer _idSerializer;

    private ICurrentUser _currentUser;

    protected override void AfterAddApplication(IServiceCollection services)
    {
        _currentUser = Substitute.For<ICurrentUser>();
        services.AddSingleton(_currentUser);
    }

    public AppMutationsTests()
    {
        _idSerializer = GetRequiredService<INodeIdSerializer>();
    }

    [Fact]
    public async Task ActivateApp()
    {
        var appId = _idSerializer.Format(nameof(App), TestAppId);
        await using var result = await ExecuteRequestAsync(b => b.SetDocument(
        $$"""
          mutation {
            activateApp(input: {id: "{{appId}}"}) {
                app {
                    enabled
                }
                errors {
                    __typename
                 }
            }
          }
          """));

        result.ShouldNotBeNull();
        result.MatchSnapshot();
    }

    [Fact]
    public async Task ChangeNameApp()
    {
        var appId = _idSerializer.Format(nameof(App), TestAppId);
        await using var result = await ExecuteRequestAsync(b => b.SetDocument(
            $$"""
              mutation {
                changeAppName(input: {
                    id: "{{appId}}"
                    name: "TestChangeName"
                })
                {
                    app {
                        name
                    }
                    errors {
                        __typename
                    }
                }
              }
              """));

        result.ShouldNotBeNull();
        result.MatchSnapshot();
    }

    [Fact]
    public async Task CreateApp()
    {
        _currentUser.Id.Returns(Guid.NewGuid());

        await using var result = await ExecuteRequestAsync(b => b.SetDocument(
            $$"""
              mutation {
                createApp(input: {
                    enabled: false,
                    name: "CreateAppTest",
                    version: {patch: 122, major: 10, minor: 20}
                }) {
                    app {
                        enabled
                        name
                        version {
                            major
                            minor
                            patch
                         }
                    }
                    errors {
                        __typename
                    }
                }
              }
              """));

        result.ShouldNotBeNull();
        result.MatchSnapshot();
    }

    [Fact]
    public async Task DeactivateApp()
    {
        var appId = _idSerializer.Format(nameof(App), TestAppId);
        await using var result = await ExecuteRequestAsync(b => b.SetDocument(
            $$"""
              mutation {
                deactivateApp(input: {id: "{{appId}}"}) {
                    app {
                        enabled
                    }
                    errors {
                        __typename
                    }
                }
              }
              """));

        result.ShouldNotBeNull();
        result.MatchSnapshot();
    }

    [Fact]
    public async Task DeleteApp()
    {
        var appId = _idSerializer.Format(nameof(App), TestAppId);
        await using var result = await ExecuteRequestAsync(b => b.SetDocument(
            $$"""
              mutation {
                deleteApp(input: {id: "{{appId}}"}) {
                    app {
                        name
                    }
                    errors {
                        __typename
                    }
                }
              }
              """));

        result.ShouldNotBeNull();
        result.MatchSnapshot();
    }
}
