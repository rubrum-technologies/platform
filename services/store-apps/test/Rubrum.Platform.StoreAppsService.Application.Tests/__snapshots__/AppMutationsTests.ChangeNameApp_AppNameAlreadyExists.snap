{
  "data": {
    "changeAppName": {
      "app": null,
      "errors": [
        {
          "code": "App:Error:NameAlreadyExists",
          "details": null,
          "logLevel": "WARNING",
          "message": "Exception of type 'Rubrum.Platform.StoreAppsService.Apps.AppNameAlreadyExistsException' was thrown.",
          "name": "Тестовое приложение dataseed",
          "__typename": "AppNameAlreadyExistsError"
        }
      ]
    }
  }
}
